### PR TITLE
ref(deploy): support latest hippo

### PR DESCRIPTION
**Pull request by michelleN**
+ updates hippo deps
+ updates deploy to use client logic from
hippo cli repo
+ includes code cleanup, error handling
+ addresses #506

Signed-off-by: Michelle Dhanani <michelle@fermyon.com>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,8 @@ dirs = "4.0"
 dunce = "1.0"
 env_logger = "0.9"
 futures = "0.3"
-hippo-openapi = "0.4"
+hippo-openapi = "0.5"
+hippo = { path = "../hippo-cli"}
```
Comment (michelleN) [inline Cargo.toml:20]: will update this when https://github.com/deislabs/hippo-cli/pull/133 is merged

**Comment by michelleN** [comment]
spin deploy with the latest changes deploys the app to Nomad but then fails on this error. Currently, debugging why this is happening, inspecting nomad job.
```
Error: invalid socket address syntax
Error: exit status: 1
Error: invalid socket address syntax
Error: exit status: 1
```

**Comment by michelleN** [comment]
Fix for Hippo was merged so we are good to go here. https://github.com/deislabs/hippo/pull/792

**Review by vdice — APPROVED** [review]
Tested this via my Nomad/Fermyon Platform setup and the `hippo register/spin deploy` flow worked great!

Note that there may be a forthcoming Hippo release fixing a few of the annotations on a scheduled job, but we can bump the version in the docs, etc., when it is released.

File: docs/content/deploying-to-hippo.md [hunk 2]
```diff
@@ -40,24 +41,13 @@ $ git checkout 64cf9334528f1975d7cbff207997d83cee4f19c2
 $ ./run_servers.sh
 ```
 
-Clone the Hippo repo locally and checkout the v0.6.2 tag. Set the `BINDLE_URL` environment variable from the `run_servers.sh` script. For now, we'll need to manuall update a few lines in the Nomad job in the project and then, run the `dotnet build` command to build the project. Change into the `src/Web` directory and use `dotnet run` to run the project.
+Clone the Hippo repo locally. The minimum version required for the Hippo server is v0.10.0. Set the `BINDLE_URL` environment variable from the `run_servers.sh` script. For now, we'll need to manuall update a few lines in the Nomad job in the project and then, run the `dotnet build` command to build the project. Change into the `src/Web` directory and use `dotnet run` to run the project.
```
Comment (vdice) [inline docs/content/deploying-to-hippo.md:44]: Should the second sentence be reworded now that users no longer need to manually update Hippo code?

Comment (michelleN) [reply docs/content/deploying-to-hippo.md:44]: good catch! will update.

**Pull request merged**